office-365-feed/operations.py

```diff
@@ -0,0 +1,275 @@
+        return result
+    except Exception as Err:
+        logger.error(str(Err))
+        raise ConnectorError(str(Err))
+
+
+def get_latest_version(config):
+    try:
+        client_request_id = str(uuid.uuid4())
+        regions_list = convert_str_list(config.get('regions'))
+        params = {"urls_list": build_urls_dict(regions_list, services_list=["All"], unique_id=client_request_id,
+                                               endpoint="version"),
+                  "verify_ssl": config.get("verify_ssl")}
+
+        resp = make_api_call(params)
+        return resp
+    except Exception as Err:
+        logger.error(str(Err))
+        raise ConnectorError(str(Err))
+
+
+def fetch_last_updated_time(config, last_pull_time):
+    region = config.get('regions')
+    latest_version_list = get_latest_version(config)
+    for version in latest_version_list:
+        if version["instance"] == region:
+            latest_time = version["latest"]
+            try:
+                old_version_ts = datetime.datetime.strptime(str(last_pull_time), "%Y-%m-%dT%H:%M:%SZ").timestamp()
+            except:
+                old_version_ts = last_pull_time
+            latest_version_ts = int(datetime.datetime.strptime(latest_time, "%Y%m%d%H").timestamp())
+            if old_version_ts < latest_version_ts:
+                return True, latest_version_list
+    return False, None
+
+
+def handle_dedup(iterable):
+    # Remove duplicate entries from result
+    try:
+        seen = set()
+        result = []
+
+        for dic in iterable:
+            key = (dic['value'])
+            if key in seen:
+                continue
+            result.append(dic)
+            seen.add(key)
+        return result
+
+    except Exception as Err:
+        logger.error(str(Err))
+        raise ConnectorError(str(Err))
+
+
+def build_response(indicator_fields, indicator_type_lower, limit=-1):
+    try:
+        iterator = make_api_call(indicator_fields)
+        result = indicator_fields["urls_list"][0]
+        # filter indicator_type specific entries
+        if not indicator_type_lower == 'both':
+            iterator = [i for i in iterator if indicator_type_lower in i]
+        indicators = []
+
+        for item in iterator:
+            if indicator_type_lower == 'both':
+                values = item.get('ips', []) + item.get('urls', [])
+            else:
+                values = item.get(indicator_type_lower)
+            if values:
+                for value in values:
+                    type_ = check_indicator_type(value)
+                    raw_data = {
+                        'value': value,
+                        'type': type_,
+                    }
+                    for key, val in item.items():
+                        if key not in ['ips', 'urls']:
+                            raw_data.update({key: val})
+
+                    raw_data['tlp'] = indicator_fields['tlp'] if indicator_fields['tlp'] else ""
```

Comment:
all these can be plummed directly in Create Record step. not needed in action output